admin/metabox/class-metabox.php

```diff
@@ -323,9 +323,14 @@ private function get_content_sections() {
 			$content_sections[] = $this->get_advanced_meta_section();
 		}
 
+		if ( ! file_exists( WPSEO_PATH . 'premium/' ) ) {
```

Comment:
Please do this based on whether premium is activated, not on whether the path exists.